text/0122-sdk-hub-scope-merge.md

```diff
@@ -0,0 +1,475 @@
+
+The goal of this RFC is that 95% of users do not need to think about these concepts at all.
+In most cases, users should simply use the global methods `Sentry.xyz()` which should do the right thing.
+
+Directly interacting with the isolation scope should only be necessary for users manually instrumenting an unsupported framework or similar. This (esp. setting up a new isolation scope) can be documented in a dedicated new docs page.
+
+Accessing the current scope `getCurrentScope()` should also rarely be necessary, and be de-emphazized in the docs. Again, just use the global methods which should do the correct thing.
+
+Accessing the global scope should be documented, and is the recommended way when you want to set global data in an async way. This can be encapsulated in a dedicated new docs page.
+
+We should also add a new docs page explaining the types of scope conceptually.
+
+## Should users care about Clients?
+
+Generally speaking, for most regular use cases the client should be mostly hidden away from users.
+Users should just call `Sentry.init()`, which will setup a client under the hood. Users should generally only interact with scopes, and we should keep clients out of most public facing APIs.
+
+The client is only there to allow an escape hatch when users need to do more complex/special things, like isolating Sentry instances or multiplexing. So client APIs should be designed to _allow_ to do things that cannot be done via `Sentry.init()`, but our main focus should be on making the default experience easy to understand, which includes that users should not have to care about the concept of clients by default.
+
+## What about other Hub references?
+
+While the Hub is mainly exposed via `getCurrentHub()`, it is also used as argument or similar in many places.
+These occurences should be updated to instead take a scope or a client.
+
+## What about backwards compatibility?
+
+We should strive to provide a wrapper/proxy `getCurrentHub()` method that still exposes the key functionality to ease upgrading. E.g.:
+
+```js
+import { getScope, getClient, captureException, withScope } from "../internals";
+
+function getCurrentHub() {
+  return {
+    getClient,
+    getScope,
+    captureException,
+    withScope,
+    // ...
+  };
+}
+```
+
+Based on the SDK, we can decide to keep _everything_ in this proxy (then we can do this even in a minor release),
+or keep _most of it_ (if we do a major) - to break as little things in user land as possible.
+
+## What about globals?
+
+This RFC does not propose any concrete way to store the active scope. This is up to the concrete SDK and may behave the same way as it currently does for the hub, or differently if that makes more sense in a given scenario.
+
+# Drawbacks
+
+- This changes _a lot_ of public APIs and behavior.
+
+# Unresolved questions
+
+- How should we handle this in docs?
+- Naming: Can we come up with something better than Isolation Scope?
```

Comment:
Just throwing in some ideas: How about having a Global Scope / Root Scope / Current Scope? Or if we want to be similar to otel, naming it Context instead of scope; although it's not exactly the same as context is just a part of the scope as I understood it. Then we'd have Global Execution Context / Root Execution Context / Current Execution Context.
Not super happy with my ideas either, as root could imply some tree structure 🤷

Reply:
We also had root as a name already in earlier iterations here 😅 which was also found to not be super understandable by some folks. It's a hard naming problem 😬